doc/api/async_hooks.md

```diff
@@ -864,6 +864,31 @@ for (let i = 0; i < 10; i++) {
 }
 ```
 
+### Integrating `AsyncResource` with `EventEmitter`
+
+Event listeners triggered by an [`EventEmitter`][] may be run in a different
+execution context than the one that was active when `eventEmitter.on()` was
+called.
+
+The following example shows how to use the `AsyncResource` class to properly
+associate an event listener with the correct execution context. The same
+approach can be applied to a [`Stream`][] or a similar event-driven class.
+
+```js
+const { createServer } = require('http');
+const { AsyncResource, executionAsyncId } = require('async_hooks');
+
+const server = createServer(function(req, res) {
+  const asyncResource = new AsyncResource('request');
+  const asyncId = asyncResource.asyncId();
+  // The listener will always run in the execution context of `asyncResource`.
+  req.on('close', asyncResource.runInAsyncScope.bind(asyncResource, () => {
+    console.log(executionAsyncId()); // Prints the value of `asyncId`.
```

Comment:
Nit: maybe?
```suggestion
    console.log(`${asyncId} === ${executionAsyncId()}`);
```

Reply:
Simplified to:
```js
// Prints: true
console.log(asyncResource.asyncId() === executionAsyncId());
```

Reply:
Hm, just `true` in the output is not really clear I think something like
```js
console.log(`asyncResource.asyncId() === executionAsyncId(): ${asyncResource.asyncId() === executionAsyncId()}`)
```
would be better.

Reply:
Users will be looking at the snippet, not at the real output. So, I'd prefer to keep it as compact as possible. If someone wants to play with the code, they will be able to change the output as they want to.